lib/resty/evp.lua

```diff
@@ -225,8 +239,14 @@ local function _new_rsa(self, opts)
     end
 
     ffi_gc(evp_pkey, _C.EVP_PKEY_free)
-    if _C.EVP_PKEY_set1_RSA(evp_pkey, rsa) ~= 1 then
-        return _err()
+    if self.algo == "RSA" then
+        if _C.EVP_PKEY_set1_RSA(evp_pkey, key) ~= 1 then
+           return _err()
+        end
+    elseif self.algo == "ECDSA" then
+        if _C.EVP_PKEY_set1_EC_KEY(evp_pkey, key) ~= 1 then
+            return _err()
+        end
     end
```

Comment:
I'm gonna add comments stream of consciousness style to reason through these changes so don't mind me.

This block looks great and opens the door for many more. Thanks for centralizing this.